woocommerce/woocommerce-gutenberg-products-block
# Filter Products By Price block: don't allow to insert negative values on inputs

This PR fixes some not unexpected behavior for the Filter Products By Price block:

### The user can insert a negative value in the inputs

Now, with this PR, the user can't insert a negative value. This is possible thanks to the prop `allowNegative`setted to `true`.


### The user can "break" the slider when inserting in the right input a value that is lower than the value on the left input
Now, with this PR, when the user inserts in the right input a value that is lower than the value on the left input, the latter is reset to 0.
This check is done thanks to `isValidMinValue`and `isValidMaxValue` and [this condition](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5123/files#diff-615d49ba906702d5afd8c7c4e079b5984633a74a1cd6eee0c1a74ef56c6a7d5bR223).



Fixes #2695 




### Testing


### Manual Testing

How to test the changes in this Pull Request:


Check out this branch

1. Make sure you have Woo Blocks plugins installed.
2. Add "All Products" block.
3. Add "Filter Products by Price" block
4. Add "Active Product filters" block
5. Click on Preview or Publish the page.
5. Go to the page having all the above block added.

Check that:

* the user can't insert in both inputs a negative number
* the user can't insert on input left a number that is greater than input on the right
* if the user inserts on the input on the right a number that is lower than input on the left, the component sets to 0 the minimum price


### Changelog
> Fixed Filter Products By Price Block: the user can't insert a negative value.
